oak/server/rust/oak_loader/src/main.rs

```diff
@@ -108,7 +93,25 @@ fn main() {
         },
     };
 
-    if let Err(e) = start_runtime(application, runtime_config) {
-        error!("failed to start Oak Runtime: {}", e);
+    // Start the Runtime from the given config.
+    info!("starting Runtime");
+    let (runtime, initial_handle) = configure_and_run(app_config, runtime_config)
+        .map_err(|status| format!("status {:?}", status))?;
+    info!("initial write handle {:?}", initial_handle);
+
+    let done = Arc::new(AtomicBool::new(false));
```

Comment:
Have you considered a https://doc.rust-lang.org/stable/std/sync/struct.Condvar.html instead? Also alternatively https://docs.rs/tokio-signal/0.3.0-alpha.1/tokio_signal/index.html (using futures) though it may require a tokio executor, which we may even already have lying around. Anyway perhaps something to look into separately.

Reply:
I did consider it, but the synchronization mechanism needs to be [async-signal-safe](http://man7.org/linux/man-pages/man7/signal-safety.7.html), and (I think) the relevant `Mutex, Condvar` combination isn't.

Reply:
Interesting, thanks for the link! Is this something that the type system would prevent? If not, perhaps worth adding a comment so that the next person doesn't just do something like that without realising its risks.

Reply:
I think a link to the doc you posted here would be useful; naively, after reading that comment I would still think that a Mutex would be fine.

Reply:
Okey dokey, I'll make it more explicit.